numba/_helperlib.c

```diff
@@ -1327,6 +1327,80 @@ numba_reset_list_private_data(PyListObject *listobj)
         listobj->allocated = PyList_GET_SIZE(listobj);
 }
 
+
+static PyObject *private_data_dict = NULL;
+
+static PyObject *
+_get_private_data_dict(void)
+{
+    if (private_data_dict == NULL)
+        private_data_dict = PyDict_New();
+    return private_data_dict;
+}
+
+NUMBA_EXPORT_FUNC(void)
+numba_set_set_private_data(PyObject *setobj, void *ptr)
```

Comment:
How do you feel about altering these names to `numba_setter_for_set_....` or `numba_set_private_data_setter` purely because `set` is overloaded ? For a minute I thought the `get_set` below was a function for `get` OR `set` and was looking for a flag to tweak the behaviour, in context it obviously makes sense as the `get` function for `set`s, but out of context and in the future I wonder how clear it will be. I guess the other option is to document it as such?